lib/axiom/fs/base/file_system.js

```diff
@@ -205,11 +205,16 @@ FileSystem.prototype.readFile = function(
   var whence = opt_whence || SeekWhence.Begin;
 
   return this.createOpenContext(path, openMode).then(
-      function(cx) {
-        return cx.open().then(function() {
-          return cx.read(offset, whence, dataType);
-        });
-  });
+    function(cx) {
+      return OpenContext.scope(cx, function() {
+        return cx.open().then(
+          function() {
```

Comment:
Well, the style for promises is a little bit all over the place at this point. I was trying to follow the style of `Ephemeral.js` where there is no extra +2 indent (e.g. https://github.com/chromium/axiom/blob/master/lib/axiom/core/ephemeral.js#L83).